deps/rabbit_common/src/rabbit_misc.erl

```diff
@@ -1265,8 +1265,14 @@ sequence_error([T])                      -> T;
 sequence_error([{error, _} = Error | _]) -> Error;
 sequence_error([_ | Rest])               -> sequence_error(Rest).
 
-check_expiry(N) when N < 0                 -> {error, {value_negative, N}};
-check_expiry(_N)                           -> ok.
+check_expiry(N)
+  when N < 0 ->
+    {error, {value_negative, N}};
+check_expiry(N)
+  when N > 3_153_600_000_000 -> %% 100 years in milliseconds
```

Comment:
Should we use something like 10 years? Some of us won't be around to test
the 100 year expiration so I think we should treat it as an "invalid value" for validation purposes.

Reply:
🙂  I reduced it to 10 years.